File: 1075. 链表元素分类(25).py
address = {}
point = {}
start,N,K = [int(i) for i in input().split(" ")]
for i in range(N):
    Address,Data,Next = [int(item) for item in input().split(" ")]
    address[Address] = Data
    point[Data] = Next

less_than_zero = []
zero_to_K = []
more_than_K = []

while start!=-1:
    if address[start] < 0:
        less_than_zero.append(start)
    elif address[start] >= 0 and address[start] <= K:
        zero_to_K.append(start)
    else:
        more_than_K.append(start)
    start = point[address[start]]

lst = less_than_zero + zero_to_K + more_than_K
for i in range(len(lst)-1):
    print("%05d"%lst[i],address[lst[i]],"%05d"%lst[i+1])
print("%05d" %lst[-1],address[lst[-1]],-1)

File: 1075. 链表元素分类(25).cpp
#include <iostream>  
#include <vector>  
using namespace std;
struct node {  
    int data, next;  
}list[100000];  
vector<int> v[3];  

int main() {  
    int start,N,K;  
    scanf("%d%d%d",&start,&N,&K);
    for(int i=0;i<N;i++){  
        int address;
        scanf("%d",&address);  
        scanf("%d%d",&list[address].data,&list[address].next);
    } 
    while(start!=-1){  
        int data = list[start].data;  
        if(data < 0)  
            v[0].push_back(start);  
        else if(data >= 0 && data <= K)  
            v[1].push_back(start);  
        else  
            v[2].push_back(start);
        start = list[start].next;  
    }   
    int flag = 0;
    for(int i=0;i<3;i++){
        for(int j=0;j<v[i].size();j++){
            if(flag==0){
                printf("%05d %d ", v[i][j], list[v[i][j]].data);
                flag = 1;
            } else {
                printf("%05d\n%05d %d ", v[i][j], v[i][j], list[v[i][j]].data);
            }
        }
    }
    printf("-1");
    return 0;  
}  
